src/main.ts

```diff
@@ -20,27 +24,54 @@ async function run(): Promise<void> {
         if (currentVersion === null)
             return
 
-        const lastCommitMessage = await getLastCommitMessage()
-        if (lastCommitMessage === null)
-            return
-
-        const releaseType = getReleaseTypeFromCommitMessage(lastCommitMessage)
+        let releaseType: string | null = 'non-release'
+        if (releaseSinceLastTag) {
+            core.info('Inferring release type from contents of the release')
+            releaseType = await getReleaseTypeFromCommitsSinceLastTag()
+        } else {
+            const lastCommitMessage = await getLastCommitMessage()
+            if (lastCommitMessage === null)
+                return
+            core.info(`Inferring release type from last commit message: ${lastCommitMessage}`)
+            const releaseType = getReleaseTypeFromCommitMessage(lastCommitMessage)
+        }
 
-        // If the commit isn't of type `feat` or `fix`, we don't want to bump the version
-        if (releaseType !== null) {
+        // If we're creating a release we want to bump the version
+        if (releaseType !== null && releaseType !== 'non-release') {
+            core.info(`New release version should be type ${releaseType}`)
+            core.setOutput('release-type', releaseType)
             const nextVersion = isReleaseCandidate
-                ? getNextVersion({ currentVersion, releaseType })
+                ? getNextVersion({ currentVersion, releaseType: releaseType as ReleaseType })
```

Comment:
You won't need the type casting if you fix the type on let releaseType

Reply:
Done